### ISSUE TITLE
Cuba Doesnt start (HA addon)

### ISSUE THREAD

**Issue by jeroenvdd** [thomasddn/qbha#4]
**Expected behavior**
starting the addon

**QBHA version:**
v0.1.0

**Home Assistant:**
Installation type:  HAOS
Version: latest

**Additional context**
I receive this error:
```
2023-11-07 17:35:03,435 qbha.Qbha INFO - MQTT client 'qbha' connecting to core-mosquitto:1883 with user 'addons'.
2023-11-07 17:36:24,990 qbha ERROR - [Errno 2] No such file or directory: '/data/qbusconfig.json'
Traceback (most recent call last):
  File "/app/main.py", line 74, in <module>
    qbha.start()
  File "/app/Qbha.py", line 29, in start
    client.loop_forever()

```

but I see no variables
![image](https://github.com/thomasddn/qbha/assets/6975276/d2d149a7-8ab0-4f63-864f-bfd1608364be)

Shouldn't this contain the qbus ip / pw / sn etc?

or isn't this available for HA yet and do I have to take a look at the data folder?

**Comment by thomasddn** [comment]
A few things to note here:
1. There was a permission error in v0.1 related to the data directory. Can you give it a shot with v0.2?
2. The `qbusconfig.json` file is generated once the "Qbus MQTT" addon starts. So restart Qbus MQTT if this file is missing. In the next version, a log entry will make this clear.
3. Qbus IP etc is not required, the Qbus MQTT addon detects the Qbus controller automatically.

**Comment by jeroenvdd** [comment]
I Am rrunning the 0.2
I'll try a clean reboot, it might be an error due to my active setup being connected already

**Comment by thomasddn** [comment]
In your opening post, you mentioned v0.1.0.

To be sure, I have tested it in a clean HAOS test environment with the following addons (all with default settings):
* MQTT
* QBHA v0.2.0
* Qbus MQTT Gateway v0.1.0

The log of QBHA should look similar to this:

```
[11:10:10] INFO: Retrieving MQTT settings from MQTT addon.
2023-11-10 11:10:11,845 qbha INFO - Starting qbha v0.2.0.
2023-11-10 11:10:11,848 qbha.Qbha INFO - MQTT client 'qbha' connecting to core-mosquitto:1883 with user 'addons'.
2023-11-10 11:10:21,513 qbha.Subscribers.QbusControllerStateSubscriber INFO - Requesting controller firmware update.
2023-11-10 11:11:03,705 qbha.Subscribers.QbusConfigSubscriber INFO - New Qbus config, updating Home Assistant entities.
```

Note the logging of the version. If you are not seeing that log entry, then you are still running v0.1.0 of QBHA.

**Comment by thomasddn** [comment]
Can you try to restart the **Qbus MQTT Gateway** addon? Leave the QBHA addon running. After the gateway has been restarted, wait a few minutes. After that, turn a light on or off and check the logs.

**Comment by jeroenvdd** [comment]
I'll try the only issue is the QBHA addon boots and shuts down so I 'can't keep it running'

**Comment by thomasddn** [comment]
> I'll try the only issue is the QBHA addon boots and shuts down so I 'can't keep it running'

Oh, right. This file is only generated when the Qbus MQTT Gateway pushes some Qbus specific topics. In an upcoming version, I added a 'file exists' check before trying to read from it. Maybe I can push it later today.

> If the task is succesfull how do the entities get created? under one MQTT device ?

You should see topics under the `homeassistant` topic, e.g. `homeassistant/light/qbus_UL1_28/config`.

**Comment by thomasddn** [comment]
I just released v0.3.0. Can you try with this one? It should at least resolve shutting down the addon when the file doesn't exist.

If it still doesn't work, then try this:
1. Shut down all Qbus related third-party software. Also other stuff you might have from the Qbus forum (e.g. nodered flows, docker containers, Raspberry Pi installation, openhab, ...). Do not touch the Qbus Controller.
2. Delete the entire `cloudapp` topic on the MQTT broker. It will be recreated later.
3. Start the QBHA addon (v0.3.0).
4. Start the Qbus MQTT Gateway addon.
5. Wait a bit and check logs.

If it is still broken, then 🤯. Nah, then we'll keep on investigating.

**Comment by jeroenvdd** [comment]
![image](https://github.com/thomasddn/qbha/assets/6975276/7ad65f51-da34-4800-bc40-7f9778ea62d2)

lijkt ok te zijn :)
